full-service/src/service/burn_transaction.rs

```diff
@@ -0,0 +1,337 @@
+                account_id_hex.to_string(),
+                self.ledger_db.clone(),
+                self.fog_resolver_factory.clone(),
+                self.logger.clone(),
+            );
+
+            let amount =
+                Amount::try_from(amount).map_err(BurnTransactionServiceError::InvalidAmount)?;
+            let default_fee_token_id = amount.token_id;
+
+            builder.add_recipient(burn_address(), amount.value, amount.token_id)?;
+
+            if let Some(tombstone) = tombstone_block {
+                builder.set_tombstone(tombstone.parse::<u64>()?)?;
+            } else {
+                builder.set_tombstone(0)?;
+            }
+
+            let fee_token_id = match fee_token_id {
+                Some(t) => TokenId::from(t.parse::<u64>()?),
+                None => default_fee_token_id,
+            };
+
+            let fee_value = match fee_value {
+                Some(f) => f.parse::<u64>()?,
+                None => *self.get_network_fees().get(&fee_token_id).ok_or(
+                    BurnTransactionServiceError::DefaultFeeNotFoundForToken(fee_token_id),
+                )?,
+            };
+
+            builder.set_fee(fee_value, fee_token_id)?;
+
+            builder.set_block_version(self.get_network_block_version());
+
+            if let Some(inputs) = input_txo_ids {
+                builder.set_txos(&conn, inputs)?;
+            } else {
+                let max_spendable = if let Some(msv) = max_spendable_value {
+                    Some(msv.parse::<u64>()?)
+                } else {
+                    None
+                };
+                builder.select_txos(&conn, max_spendable)?;
+            }
+
+            let mut memo_data = [0; BurnRedemptionMemo::MEMO_DATA_LEN];
+
+            if let Some(redemption_memo) = redemption_memo {
+                hex::decode_to_slice(&redemption_memo, &mut memo_data)
+                    .map_err(|_| BurnTransactionServiceError::InvalidNumberOfBytes)?;
+            }
+
+            let mut memo_builder = BurnRedemptionMemoBuilder::new(memo_data);
+            memo_builder.enable_destination_memo();
+            let tx_proposal: TxProposal =
+                builder.build(Some(Box::new(memo_builder)), &conn).unwrap();
+
+            TransactionLog::log_built(tx_proposal.clone(), "".to_string(), account_id_hex, &conn)?;
+
+            Ok(tx_proposal)
+        })
+    }
+
+    fn build_unsigned_burn_transaction(
+        &self,
+        account_id_hex: &str,
+        amount: &AmountJSON,
+        redemption_memo: Option<String>,
+        input_txo_ids: Option<&Vec<String>>,
+        fee_value: Option<String>,
+        fee_token_id: Option<String>,
+        tombstone_block: Option<String>,
+        max_spendable_value: Option<String>,
+    ) -> Result<(UnsignedTx, FullServiceFogResolver), BurnTransactionServiceError> {
+        let conn = self.wallet_db.get_conn()?;
+        transaction(&conn, || {
+            let mut builder = WalletTransactionBuilder::new(
```

Comment:
There seem to be a lot of duplication in the signed and unsigned flows. It is possible to unify them into a method that returns the `WalletTransactionBuilder` and then just the last part of either `.build()` or `.build_unsigned` could be different?